astronomer/providers/databricks/triggers/databricks.py

```diff
@@ -82,8 +84,30 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
                             }
                         )
                     else:
-                        error_message = f"{self.task_id} failed with terminal state: {run_state}"
-                        yield TriggerEvent({"status": "error", "message": str(error_message)})
+                        if run_state.result_state == "FAILED":
+                            task_run_id = None
+                            if "tasks" in run_info:
+                                for task in run_info["tasks"]:
+                                    if task.get("state", {}).get("result_state", "") == "FAILED":
+                                        task_run_id = task["run_id"]
+                            if task_run_id is not None:
+                                run_output = await hook.get_run_output_response(task_run_id)
+                                if "error" in run_output:
+                                    notebook_error = run_output["error"]
+                                else:
+                                    notebook_error = run_state.state_message
+                            else:
+                                notebook_error = run_state.state_message
+                            error_message = (
+                                f"{self.task_id} failed with terminal state: {run_state} "
+                                f"and with the error {notebook_error}"
+                            )
+                        else:
+                            error_message = (
+                                f"{self.task_id} failed with terminal state: {run_state} "
+                                f"and with the error {run_state.state_message}"
+                            )
+                        yield TriggerEvent({"status": "error", "message": error_message})
```

Comment:
I don't like so deep nesting, can we simplify this

Reply:
ok will simplify this

Reply:
Fixed it